src/librustdoc/lib.rs

```diff
@@ -862,6 +862,10 @@ fn main_args(at_args: &[String]) -> MainResult {
                     }),
                 }
             })
-        })
+        })?;
+        // Avoid crashing rustdoc in the presence of delayed bugs, emitting suboptimal docs is better
+        // than ICEing.
```

Comment:
Please link to https://doc.rust-lang.org/beta/rustdoc/advanced-features.html#interactions-between-platform-specific-docs here so it's clear why rustdoc could see invalid code.